src/Analyzers/CSharp/Analyzers/UsePatternMatching/CSharpUseNotPatternDiagnosticAnalyzer.cs

```diff
@@ -64,28 +64,46 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
             if (!styleOption.Value)
                 return;
 
-            // Look for the form: !(x is Y y)
-            if (!(node is PrefixUnaryExpressionSyntax
+            // Look for the form: !(x is Y y) and !(x is const)
+            if (node is PrefixUnaryExpressionSyntax
                 {
                     Operand: ParenthesizedExpressionSyntax
                     {
                         Expression: IsPatternExpressionSyntax
                         {
-                            Pattern: DeclarationPatternSyntax,
+                            Pattern: DeclarationPatternSyntax or ConstantPatternSyntax
                         } isPattern,
                     },
-                } notExpression))
+                } notExpression)
             {
+                syntaxContext.ReportDiagnostic(DiagnosticHelper.Create(
+                    Descriptor,
+                    isPattern.IsKeyword.GetLocation(),
+                    styleOption.Notification.Severity,
+                    ImmutableArray.Create(notExpression.GetLocation()),
+                    properties: null));
                 return;
```

Comment:
```suggestion
```

(and make into 'else if')

Reply:
How do you like if I use early check for `!(...)` and then use switch expression to get location of `is` keyword for every case to get rid of duplicate `ReportDiagnostic` call instead of `if ... else if ...`? Something like this:

```c#
if (node is not PrefixUnaryExpressionSyntax { Operand: ... parenthesizedExpression } notExpression)
    return;

var isKeywordLocation = parenthesizedExpression.Expression switch
{
    IsPatternExpressionSyntax { ... } isPattern => isPattern.IsKeyword.GetLocation(),

    BinaryExpressionSyntax { ... } isExpression => isExpression.OperatorToken.GetLocation(),

    _ => null
};

if (isKeywordLocation is null)
    return;

syntaxContext.ReportDiagnostic(DiagnosticHelper.Create(
    Descriptor,
    isKeywordLocation,
    styleOption.Notification.Severity,
    ImmutableArray.Create(notExpression.GetLocation()),
    properties: null));
 ```

Reply:
Yup!  That looks good to me!